components/filemanager/toolbar.md

```diff
@@ -10,36 +10,403 @@ position: 15
+| `FileManagerToolBarSortDirectionTool` | Renders a ButtonGroup for the sort direction. |
+| `FileManagerToolBarSortTool` | Renders the SplitButton for choosing which field to sort by. |
+| `FileManagerToolBarFileViewTool` | Renders a ButtonGroup to toggle between Grid and ListView views. |
+| `FileManagerToolBarSpacer` | Renders a spacer element. Use that if you need to add some space between the declared tools. |
+| `FileManagerToolBarViewDetailsTool` | Renders a Switch to toggle the Preview Pane for the selected file. |
+| `FileManagerToolBarSearchTool` | Renders a search box. | 
+
+### Custom tools
+
+To create a custom tool, declare a `<FileManagerToolBarCustomTool>` inside `<FileManagerToolBar>` tag. Place the content of your tool inside the `<FileManagerToolBarCustomTool>` tag.
+
+<div class="skip-repl"></div>
+
+````CSHTML
+<TelerikFileManager Data="@FileManagerData">
+                   <FileManagerToolBar>
+                       <FileManagerToolBarCustomTool>
+                           <TelerikButton>Custom tool</TelerikButton>
+                       </FileManagerToolBarCustomTool>
+                   </FileManagerToolBar>
+</TelerikFileManager>
+````
+
+### Example
+
+The following example demonstrates the customization of the FileManager Toolbar. The Toolbar includes some built-in and custom tools.
+
+````CSHTML
+@using System.IO
+@inject IJSRuntime jsInterop
+
+<TelerikFileManager Data="@FileManagerData"
+                    @bind-Path="@DirectoryPath"
+                    Height="400px"
+                    OnCreate="@OnCreateHandler"
+                    OnUpdate="@OnUpdateHandler"
+                    OnModelInit="@OnModelInitHandler"
+                    OnDownload="@OnDownloadHandler"
+                    OnDelete="@OnDeleteHandler">
+                   <FileManagerToolBar>
+                       <FileManagerToolBarSearchTool/>
+                       <FileManagerToolBarFileViewTool/>
+                       <FileManagerToolBarSpacer/>
+                       <FileManagerToolBarCustomTool>
+                           <TelerikButton Icon="@SvgIcon.Copy" OnClick="@CopyPath">Copy path</TelerikButton>
+                       </FileManagerToolBarCustomTool>
+                   </FileManagerToolBar>
+</TelerikFileManager>
+
+<!-- Move JavaScript code to a separate JS file in production -->
+<script suppress-error="BL9992">
+    function copyPath(pathValue) {
+        navigator.clipboard.writeText(pathValue);
+    }
+</script>
+
+@code {
+    private List<FlatFileEntry> FileManagerData = new List<FlatFileEntry>();
+
+    private string RootPath { get; set; } = "root-folder-path";
+    
+    private string DirectoryPath { get; set; } = "root-folder-path";
+
+    private async Task CopyPath()
+    {
+        await jsInterop.InvokeVoidAsync("copyPath", DirectoryPath);
+    }
+
+    private async Task OnCreateHandler(FileManagerCreateEventArgs args)
+    {
+        // the new item data is hardcoded for the purpose of the example
```

Comment:
* Format each complete sentence as such: start it with a capital letter and end it with the corresponding punctuation.
* Use "hard-code" instead of "hardcode".
```suggestion
        // The new item data is hard-coded for the purpose of the example.
```